ports/geckolib/string_cache/lib.rs

```diff
@@ -214,7 +223,7 @@ impl fmt::Debug for Atom {
 
 impl fmt::Display for Atom {
     fn fmt(&self, w: &mut fmt::Formatter) -> fmt::Result {
-        for c in char::decode_utf16(self.as_slice().iter().cloned()) {
+        for c in self.chars() {
```

Comment:
I actually wanted @SimonSapin to remove this in #12571, because it makes it too easy to accidentally trigger a slow stringification. Can you revert this line and remove |pub fn chars|? I'd also be ok with chars() being non-public with a comment indicating that it's slow and shouldn't be exposed.